api/src/main/java/jakarta/enterprise/lang/model/AnnotationInfo.java

```diff
@@ -43,34 +57,45 @@ default boolean isRepeatable() {
     }
 
     /**
-     * Whether this annotation has an attribute with given {@code name}.
+     * Whether this annotation has a member with given {@code name}.
      *
-     * @param name attribute name
-     * @return whether this annotation has an attribute with given {@code name}
+     * @param name member name, never {@code null}
+     * @return whether this annotation has a member with given {@code name}
+     * @throws java.lang.IllegalArgumentException if the argument is {@code null}
      */
-    boolean hasAttribute(String name);
+    boolean hasMember(String name);
 
     /**
      * Value of this annotation's attribute with given {@code name}.
-     * TODO what if it doesn't exist? null, exception, or change return type to Optional
      *
-     * @param name attribute name
-     * @return value of this annotation's attribute with given {@code name}
+     * @param name attribute name, never {@code null}
+     * @return value of this annotation's attribute with given {@code name} or {@code null} if it doesn't exist.
+     * @throws java.lang.IllegalArgumentException if the argument is {@code null}
```

Comment:
WDYT about the distintion between `NullPointerException` and `IllegalArgumentException` for `null` arguments? I personally like `IllegalArgumentException` more, but the convention seems to be (and Effective Java advises the same) to use NPE.

On the other hand, if the rest of CDI uses IAE, then I'd say we follow suit.

Reply:
Perhaps `NullPointerException` is fine, consistency is the key. I have tended to use IAE but then again methods like `Objects.requireNonNull` throw NPE